python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/graphql_helpers.py

```diff
@@ -56,37 +70,55 @@ def parse_tags(entity: dict[str, Any]) -> list[str]:
     for tag in outer_tags.get("tags", []):
         properties = tag["tag"]["properties"]
         if properties:
-            tags.append(properties["name"])
+            tags.append(
+                TagRef(
+                    display_name=properties["name"],
+                    id=tag["tag"]["urn"],
```

Comment:
⛏️ using `.get` calls rather than key lookups
```suggestion
        properties = tag.get("tag", {}).get("properties", {})
        if properties:
            tags.append(
                TagRef(
                    display_name=properties.get("name", ""),
                    id=tag.get("tag", {}).get("urn", ""),
```